ReactAndroid/src/test/java/com/facebook/react/animated/NativeAnimatedModuleTest.java

```diff
@@ -0,0 +1,134 @@
+  private List<UIBlock> mUIBlocks;
+
+  private void flushUIBlocks() {
+    mNativeAnimatedModule.willDispatchViewUpdates(mUIManager);
+    for (UIBlock block : mUIBlocks) {
+      block.execute(mock(NativeViewHierarchyManager.class));
+    }
+    mUIBlocks.clear();
+  }
+
+  @Before
+  public void setUp() {
+    ReactApplicationContext context = mock(ReactApplicationContext.class);
+    mUIManager = mock(UIManagerModule.class);
+    PowerMockito.doReturn(mock(EventDispatcher.class)).when(mUIManager).getEventDispatcher();
+    PowerMockito.doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws Throwable {
+        mUIBlocks.add(invocation.getArgumentAt(0, UIBlock.class));
+        return null;
+      }
+    }).when(mUIManager).addUIBlock(any(UIBlock.class));
+    PowerMockito.doAnswer(new Answer<Void>() {
+      @Override
+      public Void answer(InvocationOnMock invocation) throws Throwable {
+        mUIBlocks.add(0, invocation.getArgumentAt(0, UIBlock.class));
+        return null;
+      }
+    }).when(mUIManager).prependUIBlock(any(UIBlock.class));
+    PowerMockito.doReturn(mUIManager).when(context).getNativeModule(UIManagerModule.class);
+
+    mNativeAnimatedModule = new NativeAnimatedModule(context);
+    mNativeAnimatedModule.initialize();
+    mNodeManager = mock(NativeAnimatedNodesManager.class);
+    mNativeAnimatedModule.setNodesManager(mNodeManager);
+    mUIBlocks = new ArrayList<>();
+  }
+
+  @Test
+  public void testNodeManagerSchedulingWaitsForFlush() {
+    // Make sure operations are not executed until UIManager has flushed it's queue.
+    mNativeAnimatedModule.connectAnimatedNodeToView(1, 1000);
+    verify(mNodeManager, never()).connectAnimatedNodeToView(anyInt(), anyInt());
+    flushUIBlocks();
+    verify(mNodeManager, times(1)).connectAnimatedNodeToView(anyInt(), anyInt());
+  }
+
+  @Test
+  public void testNodeManagerSchedulingOperation() {
+    UIBlock otherBlock = mock(UIBlock.class);
+    mUIManager.addUIBlock(otherBlock);
+    mNativeAnimatedModule.connectAnimatedNodeToView(1, 1000);
+    flushUIBlocks();
+
+    // Connect should be called after other UI operations.
+    InOrder inOrder = inOrder(mNodeManager, otherBlock);
+    inOrder.verify(otherBlock, times(1)).execute(any(NativeViewHierarchyManager.class));
+    inOrder.verify(mNodeManager, times(1)).connectAnimatedNodeToView(1, 1000);
+  }
+
+  @Test
+  public void testNodeManagerSchedulingPreOperation() {
+    UIBlock otherBlock = mock(UIBlock.class);
+    mUIManager.addUIBlock(otherBlock);
+    mNativeAnimatedModule.disconnectAnimatedNodeFromView(1, 1000);
+    flushUIBlocks();
+
+    // Disconnect should be called before other UI operations.
+    InOrder inOrder = inOrder(mNodeManager, otherBlock);
+    inOrder.verify(mNodeManager, times(1)).disconnectAnimatedNodeFromView(1, 1000);
+    inOrder.verify(otherBlock, times(1)).execute(any(NativeViewHierarchyManager.class));
+  }
```

Comment:
Ideally we'd also test multithreading here to make sure everything gets called on the proper thread and is safe but I'm not sure how to do that.